src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -43,9 +45,356 @@ Copyright (c) 2013, Ford Motor Company
+        return false;
+      }
+      if (!ValidateLanguages(app_data[os_type][json::languages],
+                             synonyms_map)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  bool ValidateAppIdAndAppName(const smart_objects::SmartObject& app_data,
+                               std::set<std::string>& app_ids_set) const {
+    // Verify appid
+    if (!app_data.keyExists(json::appId)) {
+      LOG4CXX_WARN(logger_, error_prefix_ << "Can't find app ID in json file.");
+      return false;
+    }
+    bool result = true;
+    // Verify appid lenght
+    const std::string app_id(app_data[json::appId].asString());
+    if (app_id.length() > max_app_id_length_) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_
+                       << "An Object ID length exceeds maximum allowed ["
+                       << app_id.length() << "]>[" << max_app_id_length_
+                       << "]");
+      result = false;
+    }
+
+    // Verify that appid is unique
+    if (app_ids_set.find(app_id) != app_ids_set.end()) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_ << "An Object ID is not unigue [" << app_id
+                                 << "]");
+      result = false;
+    } else {
+      app_ids_set.insert(app_id);
+    }
+
+    // Verify that app is not registered yet
+    ApplicationSharedPtr registered_app =
+        ApplicationManagerImpl::instance()->application_by_policy_id(app_id);
+    if (registered_app) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_ << "Application with the same id: " << app_id
+                                 << " is registered already.");
+      result = false;
+    }
+    // Verify app name exist
+    if (!app_data.keyExists(json::name)) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_ << "Can't find app name in json file.");
+      return false;
+    }
+    // And app name length
+    const std::string appName(app_data[json::name].asString());
+    if (appName.length() > max_app_name_length_) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_
+                       << "Name of application exceeds maximum allowed ["
+                       << appName.length() << "]>[" << max_app_name_length_
+                       << "].");
+      result = false;
+    }
+    return result;
+  }
+
+  bool ValidateLanguages(const smart_objects::SmartObject& languages,
+                         SynonymsMap& synonyms_map) const {
+    bool result = true;
+    bool default_language_found = false;
+    const size_t languages_array_size = languages.length();
+    if (languages_array_size > max_languages_array_size_) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_ << "\"languages\" array exceeds max size ["
+                                 << languages_array_size << "]>["
+                                 << max_languages_array_size_ << "]");
+      return result = false;
```

Comment:
@LuxoftAKutsan  leftover after updating code to return false instead of result. Fixed for next commit.